core/src/main/scala/com/nvidia/spark/rapids/tool/Platform.scala

```diff
@@ -147,10 +149,13 @@ object PlatformFactory extends Logging {
       case PlatformNames.EMR_A10 => new EmrPlatform(PlatformNames.EMR_A10)
       case PlatformNames.ONPREM => new OnPremPlatform
       case p if p.isEmpty =>
-        logInfo(s"Platform is not specified. Using ${PlatformNames.ONPREM} as default.")
-        new OnPremPlatform
+        logInfo(s"Platform is not specified. Using ${PlatformFactory.getDefault.getName} " +
```

Comment:
nit: this creates a new `OnPremPlatform` instance, we can probably avoid this.

Reply:
Thanks @cindyyuanjiang. Updated the flow to use `PlatformNames.DEFAULT`.